docs/source/conf.py

```diff
@@ -339,5 +343,37 @@ def _autogenerate():
 _autogenerate()
 
 
+# CUDA target deprecation notice - can be included on any page by including:
+#
+# .. cuda-deprecated::
+
+cuda_deprecation_text = (
+    "The CUDA target built-in to Numba is deprecated, with further development "
+    "moved to the `NVIDIA numba-cuda "
+    "package <https://github.com/NVIDIA/numba-cuda>`_. Please see "
+    ":ref:`cuda-deprecation-status`."
+)
+
+
+class CudaDeprecated(SphinxDirective):
+    def run(self):
+        # Generate a warning admonition to contain the deprecation notice
+        warning = nodes.admonition(classes=["warning"])
+        warning += nodes.title(text="CUDA Target deprecation notice")
+
+        # Parse CUDA deprecation text so that the link and reference get
+        # resolved eventually
+        dummy = nodes.Element()
+        dummy.document = self.state.document
+        lines = StringList(string2lines(cuda_deprecation_text))
+        self.state.nested_parse(lines, 0, dummy)
+
+        # Add the text into the body of the warning
+        warning += dummy.children
+
+        return [warning]
+
+
 def setup(app):
     app.add_css_file('rtd-overrides.css')
+    app.add_directive('cuda-deprecated', CudaDeprecated)
```

Comment:
Should this be called `builtin-cuda-deprecated` ? I guess as it's not user facing its impact is probably minimal.

Reply:
I've left it as-is for minimal churn.

Reply:
That's fine, thanks for checking.